pydoc/developer/BENCHMARK.md

```diff
@@ -0,0 +1,470 @@
+
+-------------------------------------------------------------------- benchmark 'Strings_EncodeDecode': 2 tests ---------------------------------------------------------------------
+Name (time in ms)          Min               Max              Mean            StdDev            Median               IQR            Outliers       OPS            Rounds  Iterations
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+bench_encode[ascii]     3.4298 (1.0)      3.6450 (1.0)      3.5541 (1.0)      0.0889 (1.0)      3.5801 (1.00)     0.1436 (1.0)           2;0  281.3620 (1.0)           5           1
+bench_encode[idna]      3.4875 (1.02)     4.5255 (1.24)     3.7912 (1.07)     0.4328 (4.87)     3.5652 (1.0)      0.4869 (3.39)          1;0  263.7659 (0.94)          5           1
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+```
+
+## Benchmark Arguments
+There are a large number of commandline arguments available for configuring the benchmarks to run in a way fitting 
+to any use case.
+
+<table>
+<tr><th>Argument</th><th>Default Value</th></tr>
+<tr><th colspan="2">Description</th></tr>
+<tr>
+    <td>--benchmark-autosave</td>
+    <td>True when using "<b>make benchmark"</b></td>
+</tr>
+<tr><td colspan="2">Save the benchmark JSON results to the provided storage location</td></tr>
+<tr>
+    <td>--benchmark-storage</td>
+    <td>"file://benchmark_v2/.benchmarks" when using "<b>make benchmark"</b></td>
+</tr>
+<tr><td colspan="2">Storage location to save benchmark output JSON</td></tr>
+<tr>
+    <td>--benchmark-save</td>
+    <td>"COMMIT-ID_DATE_TIME_IS-DIRTY" example: 
+31de39be8b19c76d073a8999def6673a305c250d_20230405_145947_uncommited-changes.json</td>
+</tr>
+<tr><td colspan="2">Name to save the output JSON as. Will be saved as "counter_NAME.json"</td></tr>
+<tr>
+    <td>--size</td>
+    <td>10**8</td>
+</tr>
+<tr><td colspan="2">Problem size: length of array to use for benchmarks.</td></tr>
+<tr>
+    <td>--trials</td>
+    <td>5</td>
+</tr>
+<tr><td colspan="2">Number of times to run each test before averaging results. For tests that run as many trials as possible in 
+a given time, will be treated as number of seconds to run for.</td></tr>
+<tr>
+    <td>--seed</td>
+    <td></td>
+</tr>
+<tr><td colspan="2">Value to initialize random number generator.</td></tr>
+<tr>
+    <td>--dtype</td>
+    <td></td>
+</tr>
+<tr><td colspan="2">Dtypes to run benchmarks against. Comma separated list (NO SPACES) allowing for multiple. Accepted values: 
+int64, uint64, bigint, float64, bool, str and mixed. Mixed is used to generate sets of multiple types.</td></tr>
+<tr>
+    <td>--numpy</td>
+    <td>False</td>
+</tr>
+<tr><td colspan="2">When set, runs numpy comparison benchmarks.</td></tr>

```

Comment:
Mark this as a `stores true` so users know they do not need to provide the value.

Reply:
Added